internal/storage/firestore/firestore.go

```diff
@@ -0,0 +1,115 @@
+package firestore
+
+import (
+	"context"
+
+	"cloud.google.com/go/firestore"
+	firebase "firebase.google.com/go"
+	"github.com/kyleshepherd/discord-tk-bot/internal/storage"
+	"google.golang.org/api/iterator"
+)
+
+type (
+	KillStore struct {
+		client *firestore.Client
+	}
+)
+
+func NewKillStore(ctx context.Context, projectID string) (*KillStore, error) {
+	conf := &firebase.Config{ProjectID: projectID}
+	app, err := firebase.NewApp(ctx, conf)
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := app.Firestore(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return &KillStore{
+		client: client,
+	}, nil
+}
+
+func (s *KillStore) GetKillByID(ctx context.Context, id string) (*storage.Kill, error) {
+	dsnap, err := s.client.Collection("kills").Doc(id).Get(ctx)
+	if err != nil {
```

Comment:
It might be better to check if the err is not found see [here](https://pkg.go.dev/cloud.google.com/go/firestore#DocumentRef.Get) in the golang firestore docs and return a custom storage.ErrNotFound error if it isn't

Reply:
ah yeah good idea!